server/channels/store/localcachelayer/role_layer.go

```diff
@@ -60,15 +62,30 @@ func (s LocalCacheRoleStore) GetByNames(names []string) ([]*model.Role, error) {
 	var foundRoles []*model.Role
 	var rolesToQuery []string
 
-	for _, roleName := range names {
+	var toPass []any
+	for i := 0; i < len(names); i++ {
 		var role *model.Role
-		if err := s.rootStore.doStandardReadCache(s.rootStore.roleCache, roleName, &role); err == nil {
-			foundRoles = append(foundRoles, role)
+		toPass = append(toPass, &role)
+	}
+	errs := s.rootStore.doMultiReadCache(s.rootStore.roleCache, names, toPass)
```

Comment:
A couple of thoughts here, but they apply to the other occurrences of this pattern.

1. Not sure I fully understood the need for pointer of pointer.
2. Minor thing, but we could use `make()` and preallocate since we know the length in advance.